remote-keypair/src/openpgp_card.rs

```diff
@@ -0,0 +1,359 @@
+    pin_verified: RefCell<bool>,
+}
+
+impl TryFrom<&Locator> for OpenpgpCard {
+    type Error = openpgp_card::Error;
+
+    #[cfg(feature = "openpgp-card-pcsc")]
+    fn try_from(locator: &Locator) -> Result<Self, Self::Error> {
+        let pcsc_identifier = locator.aid.as_ref().map(|x| x.ident());
+        let backend = match pcsc_identifier {
+            Some(ident) => PcscBackend::open_by_ident(&ident, None)?,
+            None => {
+                let mut cards = PcscBackend::cards(None)?;
+                if cards.is_empty() {
+                    return Err(Self::Error::Smartcard(SmartcardError::NoReaderFoundError));
+                } else {
+                    cards.remove(0)
+                }
+            }
+        };
+
+        // Start up the long-lived OpenPgp backend connection object which will
+        // be persisted as a field in the `OpenpgpCard` struct.
+        let pgp = OpenPgp::new::<PcscBackend>(backend.into());
+
+        Ok(Self {
+            pgp: RefCell::new(pgp),
+            pin_verified: RefCell::new(false),
+        })
+    }
+
+    #[cfg(not(feature = "openpgp-card-pcsc"))]
+    fn try_from(_locator: &Locator) -> Result<Self, Self::Error> {
+        Err(Self::Error::UnsupportedFeature(
+            "openpgp-card-pcsc crate compilation disabled in solana-remote-keypair. \
+            Rebuild with `--features pcsc`."
+                .to_string(),
+        ))
+    }
+}
+
+impl Signer for OpenpgpCard {
+    fn try_pubkey(&self) -> Result<Pubkey, SignerError> {
+        let mut pgp_mut = self.pgp.borrow_mut();
+        let opt = &mut pgp_mut.transaction().map_err(|e| {
+            SignerError::Connection(format!("could not start transaction with card: {e}"))
+        })?;
+
+        // Verify smart card's PGP signing key is an ed25519 key using EdDSA
+        // and extract the pubkey as bytes.
+        let pk_material = opt
+            .public_key(openpgp_card::KeyType::Signing)
+            .map_err(|e| {
+                SignerError::Connection(format!("could not find signing keypair on card: {e}"))
+            })?;
+        let pk_bytes: [u8; 32] = match pk_material {
+            PublicKeyMaterial::E(pk) => match pk.algo() {
+                Algo::Ecc(ecc_attrs) => {
+                    if ecc_attrs.ecc_type() != EccType::EdDSA || ecc_attrs.curve() != Curve::Ed25519
+                    {
+                        return Err(SignerError::Connection(format!(
+                            "expected Ed25519 key, got {:?}",
+                            ecc_attrs.curve()
+                        )));
+                    }
+                    pk.data().try_into().map_err(|e| {
+                        SignerError::Connection(format!("key on card is malformed: {e}"))
+                    })?
+                }
+                _ => {
+                    return Err(SignerError::Connection(
+                        "expected ECC key, got RSA".to_string(),
+                    ))
+                }
```

Comment:
Similar to the error below, consider being explicit in the pattern:

```suggestion
                Algo::Rsa(_) | Alog::Unknown(_) => {
                    return Err(SignerError::Connection(
                        "expected ECC key, got RSA or an unknown algorithm".to_string(),
                    ))
                }
```